src/Durable/ActivityInvocationTracker.cs

```diff
@@ -75,6 +86,16 @@ internal class ActivityInvocationTracker
             var allTasksCompleted = completedEvents.Count == tasksToWaitFor.Count;
             if (allTasksCompleted)
             {
+                var newOrchestrationStart = context.History.FirstOrDefault(
+                    (e) => e.EventType == HistoryEventType.OrchestratorStarted &&
+                    e.Timestamp > context.CurrentUtcDateTime
```

Comment:
Thinking about this timestamp comparison: in theory, if the function is executed by a super-fast computer, the timestamps of different events may coincide. Though this may sound quite improbable, it may eventually happen. What would be the consequences of that? I'm not sure whether we will have any negative consequences or not, please let me know if you see a problem here.

Reply:
This will most likely lead to an error in the CurrentUtcDateTime during replays. For example suppose we have two OrchestratorStarted events OS1 and OS2 with the same timestamp T1. Let TS and TC denote task scheduled events and task completed events respectively, and suppose we have the first OrchestratorStarted event OS3 with timestamp T2 > T1. If the history resembles TS -> OS1 -> TC -> TS -> OS2 -> TC -> TS -> OS3 in chronological order, then upon replaying the first task completed, the CurrentUtcDateTime (representing the DateTime between OS1 and OS2) is updated to T2, when it should be T1.

Reply:
I will add a check for this in the CurrentUtcDateTime tests. I believe an appropriate way to modify this check is to use the HistoryEvent.IsProcessed instance property instead of the Timestamp.